pyhive/hive.py

```diff
@@ -131,7 +130,9 @@ def commit(self):
 
     def cursor(self, *args, **kwargs):
         """Return a new :py:class:`Cursor` object using the connection."""
-        return Cursor(self, *args, **kwargs)
+        cursor = Cursor(self, *args, **kwargs)
+        cursor.execute('USE `{}`'.format(self.database))
+        return cursor
```

Comment:
I think I see a problem here. Each request to `cursor` will execute a new `USE {}` statement. Calling it two times in a row might cause a SQL failure.